backend/zoho/zoho.go

```diff
@@ -0,0 +1,1223 @@
+	if err := o.readMetaData(context.TODO()); err != nil {
+		fs.Logf(o, "Failed to read metadata: %v", err)
+		return 0
+	}
+	return o.size
+}
+
+// setMetaData sets the metadata from info
+func (o *Object) setMetaData(info *api.Item) (err error) {
+	if info.Attributes.IsFolder {
+		return fs.ErrorNotAFile
+	}
+	o.hasMetaData = true
+	o.size = info.Attributes.StorageInfo.Size
+	o.modTime = time.Time(info.Attributes.ModifiedTime)
+	o.id = info.ID
+	return nil
+}
+
+// readMetaData gets the metadata if it hasn't already been fetched
+//
+// it also sets the info
+func (o *Object) readMetaData(ctx context.Context) (err error) {
+	if o.hasMetaData {
+		return nil
+	}
+	info, err := o.fs.readMetaDataForPath(ctx, o.remote)
+	if err != nil {
+		return err
+	}
+	return o.setMetaData(info)
+}
+
+// ModTime returns the modification time of the object
+//
+//
+// It attempts to read the objects mtime and if that isn't present the
+// LastModified returned in the http headers
+func (o *Object) ModTime(ctx context.Context) time.Time {
+	return o.modTime
+}
+
+// SetModTime sets the modification time of the local fs object
+func (o *Object) SetModTime(ctx context.Context, modTime time.Time) error {
+	return fs.ErrorCantSetModTime
+}
+
+// rename renames an object in place
+//
+// this a separate api call then move with zoho
+func (o *Object) rename(ctx context.Context, name string) (err error) {
+	item, err := o.fs.rename(ctx, o.id, name)
+	if err != nil {
+		return err
+	}
+	o.setMetaData(item)
+	return nil
+}
+
+// Open an object for read
+func (o *Object) Open(ctx context.Context, options ...fs.OpenOption) (in io.ReadCloser, err error) {
+	if o.id == "" {
+		return nil, errors.New("can't download - no id")
+	}
+	var start, end int64 = 0, o.size
+	partialContent := false
+	for _, option := range options {
+		switch x := option.(type) {
+		case *fs.SeekOption:
+			start = x.Offset
+			partialContent = true
+		case *fs.RangeOption:
+			if x.Start >= 0 {
+				start = x.Start
+				if x.End > 0 && x.End < o.size {
+					end = x.End + 1
+				}
+			} else {
+				// {-1, 20} should load the last 20 characters [len-20:len]
+				start = o.size - x.End
+			}
+			partialContent = true
+		default:
+			if option.Mandatory() {
+				fs.Logf(nil, "Unsupported mandatory option: %v", option)
+			}
+		}
+	}
+	var resp *http.Response
+	opts := rest.Opts{
+		Method:  "GET",
+		Path:    "/download/" + o.id,
+		Options: options,
+	}
+	err = o.fs.pacer.Call(func() (bool, error) {
+		resp, err = o.fs.srv.Call(ctx, &opts)
+		return shouldRetry(resp, err)
+	})
+	if err != nil {
+		return nil, err
+	}
+	if partialContent && resp.StatusCode == 200 {
+		if start > 0 {
+			// We need to read and discard the beginning of the data...
+			_, err = io.CopyN(ioutil.Discard, resp.Body, start)
+			if err != nil {
+				return nil, err
+			}
+		}
+		// ... and return a limited reader for the remaining of the data
+		return readers.NewLimitedReadCloser(resp.Body, end-start), nil
+	}
```

Comment:
Zoho ignores range requests, at least for all the common ways to do these, so I've added this workaround.

Reply:
Zoho becomes a third candidate for a future fs.Feature flag codenamed "anti-ranger, bandwidth eater" along with mailru and seafile. See discussion here https://github.com/rclone/rclone/pull/4607#issuecomment-694810408

Reply:
> Zoho becomes a third candidate for a future fs.Feature flag codenamed "anti-ranger, bandwidth eater"

:-)

I think you are right.

Reply:
I made an issue about this: https://github.com/rclone/rclone/issues/4733